hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java

```diff
@@ -47,22 +45,21 @@
 public class EntityNum implements Comparable<EntityNum> {
 	public static final EntityNum MISSING_NUM = new EntityNum(0);
 
-	private final int value;
+	private final long value;
```

Comment:
When we replace this with a `long`, imo we should simultaneously migrate the `number` fields of all the types that appear as values in a `MerkleMap` with `EntityNum` key. (For example, it's no longer _technically_ safe for `MerkleAccountState#number` to be an `int`---although in practice we could get away with this for a very long time.)

Reply:
I'm wondering if the int vs num issue might be causing an issue. Maybe I should focus on migrating this to long later on, but not here.. Let me revert the changes to long.

Reply:
Reverted EntityNum changes and will postpone this to subsequent release. Hopefully, this fixes the issues seen.